osu.Game.Rulesets.Osu/Skinning/Default/KiaiFlash.cs

```diff
@@ -37,7 +38,7 @@ protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint,
             Child
                 .FadeTo(flash_opacity, EarlyActivationMilliseconds, Easing.OutQuint)
                 .Then()
-                .FadeOut(timingPoint.BeatLength - fade_length, Easing.OutSine);
+                .FadeOut(Math.Max(0, timingPoint.BeatLength - fade_length), Easing.OutSine);
```

Comment:
The diagnosis of the issue appears to be correct and the fix does seem to work. It's a bit wasteful that `KiaiFiash` is going to be allocating transforms at that high of a BPM but I'm not sure how to handle this cleanly in a different manner.

Not sure the fade-out should be instant though, it may look wrong when the kiai section ends. Maybe this should be something like

```suggestion
                .FadeOut(Math.Max(fade_length, timingPoint.BeatLength - fade_length), Easing.OutSine);
```

to avoid zero-duration transforms? Not sure, going to see what other reviewers say.

Reply:
Can agree with this. I think we can deal with any performance overhead separately, as it likely applies to more similar cases of the `OnNewBeat` usage than just this one.